Sources/OpenAPIRuntime/URICoder/Parsing/URIParser.swift

```diff
@@ -61,13 +61,15 @@ extension URIParser {
             switch configuration.style {
             case .form: return [:]
             case .simple: return ["": [""]]
+            case .deepObject: return [:]
             }
         }
         switch (configuration.style, configuration.explode) {
         case (.form, true): return try parseExplodedFormRoot()
         case (.form, false): return try parseUnexplodedFormRoot()
         case (.simple, true): return try parseExplodedSimpleRoot()
         case (.simple, false): return try parseUnexplodedSimpleRoot()
+        case (.deepObject, _): return try parseExplodedDeepObjectRoot()
```

Comment:
My reading of the OAS is that `deepObject` is only valid with `explode: true`[^0]. If that's the case, we should probably guard against that somehow.

[^0]: https://spec.openapis.org/oas/latest.html#style-examples

Reply:
Yeap you are right, that's the case. Should I throw an error instead of ignoring the explode value?

Reply:
Yes please
```suggestion
        case (.deepObject, true): return try parseExplodedDeepObjectRoot()
        case (.deepObject, false): // throw an error here
```